nifpga/tests/test_FXP.py

```diff
@@ -0,0 +1,350 @@
+        self.set_register_attributes()
+
+    def set_register_attributes(self):
+        self._name = "MockFxpRegister"
+        self._radix_point = self._calculate_radix_point()
+        self._offset = 0  # Not needed for mock
+        self._datatype = DataType.FXP  # FXP is always DataType.FXP
+        self._ctype_type = self._datatype._return_ctype()
+        self._access_may_timeout = False  # Does not affect FXP any different
+        self._internal = False  # Cant be internal
+        self._is_array = False  # This mock is always a single FXP
+
+
+""" These are a couple of arbitrary binary strings that the following unit
+to test a non random value.
+"""
+binary_string_16bit = '1110010010010110'  # 2's comp = 0001101101101010
+binary_string_32bit = '01000101100100100011000100001100'
+
+
+class FXPRegister16bitWord16bitInteger(unittest.TestCase):
+    def setUp(self):
+        self.testRegister = MockFxpRegister(signed=False,
+                                            enableOverflowStatus=False,
+                                            word_length=16,
+                                            integer_word_length=16)
+
+    def test_converting_binary_into_decimal(self):
+        value = (int(binary_string_16bit, 2))
+        actual = self.testRegister._convert_from_read_value_to_decimal(
+            value)
+        """ The expected value should be equal to
+        2^(15) + 2^(14) + 2^(13) + 2^(10) + 2^(7) + 2^(4) + 2^(2) + 2^(1)
+        """
+        expected = Decimal(58518)
+        self.assertEqual(expected, actual)
+
+    def test_converting_user_data_into_binary(self):
+        value = Decimal(58518)
+        actual = self.testRegister._convert_data_to_binary_fxp(value)
+
+        expected = binary_string_16bit
+        self.assertEqual(expected, actual)
+
+    def test_converting_user_data_too_small_warns_and_coerces_zero(self):
+        value = Decimal(-6)  # Arbitrary negative number
+        with warnings.catch_warnings(record=True) as recordedWarning:
+            self.assertEqual(0, len(recordedWarning))
+            actual = self.testRegister._convert_data_to_binary_fxp(value)
+            self.assertEqual(1, len(recordedWarning))
+            expected = '0'.zfill(16)
+            self.assertEqual(expected, actual)
+
+
+class FXPRegister16bitWord16bitIntegerSigned(unittest.TestCase):
+    def setUp(self):
+        self.testRegister = MockFxpRegister(signed=True,
+                                            enableOverflowStatus=False,
+                                            word_length=16,
+                                            integer_word_length=16)
+
+    def test_converting_binary_into_decimal(self):
+        value = int(binary_string_16bit, 2)
+        actual = self.testRegister._convert_from_read_value_to_decimal(value)
+        """ The expected value should be equal to
+        (-1)(1*(2^(12) + 2^(11) + 2^(9) + 2^(8) + 2^(6) + 2^(5) + 2^(3) + 2^(1))
+        """
+        expected = Decimal(-7018)
```

Comment:
I seem to recall some implementation code that used the bin() function, which returns a result in a sign-magnitude format. I think you stripped off the sign, so it seems like there's an opportunity for a word-size bug (you might forget to count the sign bit somewhere). Make sure you have some tests that verify your conversions work over the complete range.